diff --git a/src/main.rs b/src/main.rs
index f7ee4c5..978264b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -112,6 +112,15 @@ fn main() -> ExitCode {
         .split('\n')
         .collect();
 
+    if args.print_packages {
+        println!(
+            "Packages that contain /bin/{command}:\n{}",
+            attrs.iter().map(|a| format!("- {a}")).collect::<Box<[String]>>().join("\n")
+        );
+
+        return ExitCode::SUCCESS;
+    };
+
     let choice = if attrs.len() > 1 {
         match pick(&args.picker, &attrs) {
             Some(x) => x,
@@ -127,19 +136,7 @@ fn main() -> ExitCode {
     }
     .contains("nixpkgs=");
 
-    if args.print_package {
-        println!(
-            "Package that contains executable /bin/{}: {}",
-            command,
-            &choice.rsplit('.').last().unwrap()
-        );
-    };
-
-    if args.install {
-        Command::new("nix-env")
-            .args(["-f", "<nixpkgs>", "-iA", choice.rsplit('.').last().unwrap()])
-            .exec();
-    } else if args.shell {
+    if args.shell {
         let shell_cmd = shell::select_shell_from_pid(process::id()).unwrap_or("bash".into());
         run_command_or_open_shell(use_channel, &choice, &shell_cmd, &[], &args.nixpkgs_flake);
     } else if args.print_path {
@@ -164,10 +161,6 @@ fn main() -> ExitCode {
 #[derive(Parser)]
 #[clap(version = crate_version!(), trailing_var_arg = true)]
 struct Opt {
-    /// Install the derivation containing the executable
-    #[clap(short, long)]
-    install: bool,
-
     /// Open a shell containing the derivation containing the executable
     #[clap(short, long)]
     shell: bool,
@@ -187,9 +180,9 @@ struct Opt {
     #[clap(short, long)]
     update: bool,
 
-    /// Print the package containing the executable
-    #[clap(short = 'p', long = "print-package")]
-    print_package: bool,
+    /// Print the packages containing the executable
+    #[clap(short = 'p', long = "print-packages")]
+    print_packages: bool,
 
     /// Print the absolute path to the executable in the nix store
     #[clap(short = 'x', long = "print-path")]
