diff --git a/src/screen.cpp b/src/screen.cpp
index c22834e5d..f47b63501 100644
--- a/src/screen.cpp
+++ b/src/screen.cpp
@@ -763,7 +763,8 @@ void screen_t::update(const wcstring &left_prompt, const wcstring &right_prompt,
         size_t start = 0;
         for (const size_t line_break : left_prompt_layout.line_breaks) {
             this->write_str(left_prompt.substr(start, line_break - start));
-            if (clr_eol) {
+            // Clear the rest of the line if the printed line was shorter that the screen width
+            if (clr_eol && (line_break - start) < (this->actual.screen_width - 1)) {
                 this->write_mbs(clr_eol);
             }
             start = line_break;
