diff --git a/src/nix/run.cc b/src/nix/run.cc
index 824201fdf..d6304f3c1 100644
--- a/src/nix/run.cc
+++ b/src/nix/run.cc
@@ -135,6 +135,7 @@ struct CmdShell : InstallablesCommand, MixEnvironment
         unixPath.insert(unixPath.begin(), pathAdditions.begin(), pathAdditions.end());
         auto unixPathString = concatStringsSep(":", unixPath);
         setenv("PATH", unixPathString.c_str(), 1);
+        setenv("IN_NIX3_SHELL", "1", 1);
 
         Strings args;
         for (auto & arg : command) args.push_back(arg);
