diff --git a/lix/nix/run.cc b/lix/nix/run.cc
index dcd81037b..8487295c1 100644
--- a/lix/nix/run.cc
+++ b/lix/nix/run.cc
@@ -149,6 +149,7 @@ struct CmdShell : InstallablesCommand, MixEnvironment
         unixPath.insert(unixPath.begin(), pathAdditions.begin(), pathAdditions.end());
         auto unixPathString = concatStringsSep(":", unixPath);
         setenv("PATH", unixPathString.c_str(), 1);
+        setenv("IN_NIX3_SHELL", "1", 1);
 
         Strings args;
         for (auto & arg : command) args.push_back(arg);
