diff --git a/my_cookies/my_cookies.py b/my_cookies/my_cookies.py
index 3d7d652..3aac452 100755
--- a/my_cookies/my_cookies.py
+++ b/my_cookies/my_cookies.py
@@ -24,10 +24,11 @@ def retrieve_cookies(domain_name, keys):
         cookie_keys = [k.strip() for k in keys.split(',')]
 
     cookie_loaders = {
-        "Chrome": browser_cookie3.chrome,
+        "LibreWolf": browser_cookie3.librewolf,
+        "Firefox": browser_cookie3.firefox,
         "Chromium": browser_cookie3.chromium,
         "Brave": browser_cookie3.brave,
-        "Firefox": browser_cookie3.firefox,
+        "Chrome": browser_cookie3.chrome,
         "Edge": browser_cookie3.edge,
     }
 
